From 42e37735107b77e5fab66038c2b89ebc2fd52915 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@google.com>
Date: Mon, 10 Feb 2020 08:40:18 -0800
Subject: [PATCH 020/100] ANDROID: sdcardfs: fix up fs parsing calls

The filesystem param parsing logic has changed, so fix up sdcardfs to
use the correct api and not break the build.

Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
Change-Id: I4e8b8f0b4998a852988a710ca18736046e3d8697
---
 fs/sdcardfs/main.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/fs/sdcardfs/main.c b/fs/sdcardfs/main.c
index 818122410bfe..34b41071c379 100644
--- a/fs/sdcardfs/main.c
+++ b/fs/sdcardfs/main.c
@@ -41,7 +41,7 @@ enum sdcardfs_param {
 	Opt_err,
 };
 
-static const struct fs_parameter_spec sdcardfs_param_specs[] = {
+static const struct fs_parameter_spec sdcardfs_fs_specs[] = {
 	fsparam_u32("fsuid", Opt_fsuid),
 	fsparam_u32("fsgid", Opt_fsgid),
 	fsparam_u32("gid", Opt_gid),
@@ -57,11 +57,6 @@ static const struct fs_parameter_spec sdcardfs_param_specs[] = {
 	{}
 };
 
-static const struct fs_parameter_description sdcardfs_parameters = {
-	.name	= "sdcardfs",
-	.specs	= sdcardfs_param_specs,
-};
-
 static int sdcardfs_parse_param(struct fs_context *fc, struct fs_parameter *param)
 {
 	struct sdcardfs_context_options *fc_opts = fc->fs_private;
@@ -70,7 +65,7 @@ static int sdcardfs_parse_param(struct fs_context *fc, struct fs_parameter *para
 	struct fs_parse_result result;
 	int opt;
 
-	opt = fs_parse(fc, &sdcardfs_parameters, param, &result);
+	opt = fs_parse(fc, sdcardfs_fs_specs, param, &result);
 	if (opt < 0)
 		return opt;
 
-- 
2.17.1

